### PR TITLE
AP_Param: avoid allocating 0 bytes if no defaults

### PR DESCRIPTION
**Pull request by peterbarker**
This stopped CUAVv5 from booting

### CODE REVIEW

File: libraries/AP_Param/AP_Param.cpp [hunk 1]
```diff
@@ -2037,6 +2041,9 @@ void AP_Param::load_embedded_param_defaults(bool last_pass)
     if (!count_embedded_param_defaults(num_defaults)) {
         return;
     }
+    if (num_defaults == 0) {
+        return true;
```
Comment (guglie) [inline libraries/AP_Param/AP_Param.cpp]: I think this should be `return;`

Comment (peterbarker) [reply libraries/AP_Param/AP_Param.cpp]: Fixed, thanks.

**Comment by peterbarker** [comment]
tridge wants this moved to new/delete rather than malloc/free

**Comment by peterbarker** [comment]
This was moved to new/delete

File: libraries/AP_Param/AP_Param.cpp [hunk 2]
```diff
@@ -2027,28 +2029,26 @@ bool AP_Param::count_embedded_param_defaults(uint16_t &count)
  */
 void AP_Param::load_embedded_param_defaults(bool last_pass)
 {
-    if (param_overrides != nullptr) {
-        free(param_overrides);
-        param_overrides = nullptr;
-    }
-    
+    delete param_overrides;
```
Comment (tridge) [inline libraries/AP_Param/AP_Param.cpp]: I think this should be delete[]

**Pull request merged**